test/e2e/framework/deployment/fixtures.go

```diff
@@ -118,6 +118,9 @@ func GetPodsForDeployment(ctx context.Context, client clientset.Interface, deplo
 		// Remove hash labels from template.Labels before comparing
```

Comment:
Can you please add a new test (probably E2E) such as:
- replica set and deployment exists with only the deprecated label
- it should be possible to scale such deployment without a new label
- update the deployment and observe a new replica set with both labels
- it should be possible to rollback to the old replica set

Reply:
Hi @atiratree ! Thanks for your review! I've address the rest of your comments. And now I'm trying to write a e2e test for this pr as you suggested.

Currently I have not started writing the test code, but trying to test this manually with the cluster build from the source code. However, I encountered some problem with the very first step. 

That is, how can we create a deployment/replicaset with only the deprecated label😂. 

I tried to create a new deployment and use kubectl to edit the deployment, but there are some problems:
1. labelSelector is immutable, and the deployment will only have the new label according to this pr.
2. If we remove the new label in spec.template, the labelSelector will not match the label in spec, and the edit will fail also.

Reply:
@AxeZhan you have to first create the replica set and then the deployment. You can also assign the owner reference after you create the deployment to this rs, but that is probably not needed, just for completeness..

Reply:
> @AxeZhan you have to first create the replica set and then the deployment. You can also assign the owner reference after you create the deployment to this rs, but that is probably not needed, just for completeness..

Well, I may still need some help. @atiratree  Here's what I did:

1. create a simple deployment
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: my-deployment
spec:
  replicas: 0  # Number of desired replicas
  selector:
    matchLabels:
      app: my-app  # Label selector for the Pods
  template:
    metadata:
      labels:
        app: my-app  # Labels to match the selector
    spec:
      containers:
        - name: my-container
          image: nginx:latest  # Replace with your desired image
```
2. Create a rs with old label, and ownerReference to the existing deployment:
```
apiVersion: apps/v1
kind: ReplicaSet
metadata:
  name: my-replicaset
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: Deployment
    name: my-deployment
    uid: 9c5ddce8-1f69-463d-9def-615e3f6afb03
spec:
  replicas: 1  # Number of desired replicas
  selector:
    matchLabels:
      app: my-app  # Label selector for the Pods
      pod-template-hash: hash
  template:
    metadata:
      labels:
        pod-template-hash: hash
        app: my-app  # Labels to match the selector`
    spec:
      containers:
        - name: my-container
          image: nginx:latest  # Replace with your desired image
```
However, this seems can't actually change the replicaset I created. After applying the yaml, I use `kubectl get rs my-rs -o yaml` and get this:
```
apiVersion: apps/v1
kind: ReplicaSet
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"apps/v1","kind":"ReplicaSet","metadata":{"annotations":{},"name":"my-replicaset","namespace":"default","ownerReferences":[{"apiVersion":"apps/v1","blockOwnerDeletion":true,"controller":true,"kind":"Deployment","name":"my-deployment","uid":"9c5ddce8-1f69-463d-9def-615e3f6afb03"}]},"spec":{"replicas":1,"selector":{"matchLabels":{"app":"my-app","pod-template-hash":"hash"}},"template":{"metadata":{"labels":{"app":"my-app","pod-template-hash":"hash"}},"spec":{"containers":[{"image":"nginx:latest","name":"my-container"}]}}}}
  creationTimestamp: "2023-08-08T11:41:30Z"
  generation: 1
  name: my-replicaset
  namespace: default
  resourceVersion: "27771"
  uid: 9cde1cc7-fd2a-44ae-9838-2aea72f83f5b
spec:
  replicas: 1
  selector:
    matchLabels:
      app: my-app
      pod-template-hash: hash
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: my-app
        pod-template-hash: hash
    spec:
      containers:
      - image: nginx:latest
        imagePullPolicy: Always
        name: my-container
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
status:
  availableReplicas: 1
  fullyLabeledReplicas: 1
  observedGeneration: 1
  readyReplicas: 1
  replicas: 1
```
Still, I can't use this deployment to control the rs I created :( . The deployment is controlling a replicaset with new labels.

Reply:
I don't think you can change the RS after it is created. You need to create the RS first programmatically (calculate the hash manually and everything). And then create the deployment with the same pod template.

Reply:
Ah thanks, I tried create rs then deployment before but it failed. But now this worked😂.
I've test this manually on my cluster and it works fine, I'll work on the e2e test later.